### PR TITLE
Doc: interleave columns pandas compat

### PR DESCRIPTION
**Pull request by raybellwaves**
## Description

Add a `pandas_compat` note to `DataFrame.interleave_columns`

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -4379,7 +4379,7 @@ def query(self, expr, local_dict=None):
         1 2018-10-08
 
         .. pandas-compat::
-            **DataFrame.query**
+            :func:`pandas.DataFrame.query`
```
Comment (bdice) [inline python/cudf/cudf/core/dataframe.py:4382]: We've been just putting the API name in bold and not cross-linking it with intersphinx. I generally like using intersphinx but we'll have to see how it renders. If this change is accepted, would you be willing to make all the other `.. pandas-compat::` sections match with an intersphinx reference in a second PR?

Comment (raybellwaves) [reply python/cudf/cudf/core/dataframe.py:4382]: > If this change is accepted, would you be willing to make all the other `.. pandas-compat::` sections match with an intersphinx reference in a second PR?

Should be ok.

Any idea why the docs build was skipped? https://github.com/rapidsai/cudf/actions/runs/8423601396/job/23067371447?pr=15383

**Comment by bdice** [comment]
We'll need #15392 to merge to 24.04, then forward-merge 24.04 to 24.06, then merge 24.06 into this PR before the docs will build correctly.

**Comment by raybellwaves** [comment]
Had some issues with the pandas sphinx intermapping as well displaying a mapping in the pandas-compat directive (https://github.com/orgs/sphinx-doc/discussions/12212). I'll open back if I make any progress

**Comment by vyasr** [comment]
I'm holding off reviewing this until #15531 is in and we know that the pandas intersphinx mapping is working as expected. Happy to pick it back up again afterwards!

**Comment by raybellwaves** [comment]
@vyasr no rush. I think i want to take a look at the pandas-compat sphinx extension first

**Comment by vyasr** [comment]
OK sounds good. Now that #15531 feel free to ping again when you think this PR is ready for another look.

**Comment by raybellwaves** [comment]
> OK sounds good. Now that #15531 is merged feel free to ping again when you think this PR is ready for another look.

I think is ready for review now.

**Review by wence- — APPROVED** [review]
Looks good, thanks!

**Pull request merged**